handlers/utils/compute_utils.go

```diff
@@ -0,0 +1,1186 @@
+	if ok {
+		host.Data["clusterConnection"] = InterfaceToString(clusterConnection)
+		if strings.HasPrefix(InterfaceToString(clusterConnection), "http") {
+			caCrt, ok1 := GetFieldsIfExist(event.Data, "field", "caCrt")
+			clientCrt, ok2 := GetFieldsIfExist(event.Data, "field", "clientCrt")
+			clientKey, ok3 := GetFieldsIfExist(event.Data, "field", "clientKey")
+			// what if we miss certs/key? do we have to panic or ignore it?
+			if ok1 && ok2 && ok3 {
+				host.Data["caCrt"] = InterfaceToString(caCrt)
+				host.Data["clientCrt"] = InterfaceToString(clientCrt)
+				host.Data["clientKey"] = InterfaceToString(clientKey)
+			} else {
+				logrus.Infof("Missing certs/key [%v]for clusterConnection for connection ",
+					clusterConnection)
+			}
+		}
+	}
+	return &instance, &host
+}
+
+func IsInstanceActive(instance *model.Instance, host *model.Host) bool {
+	if IsNoOp(instance.Data) {
+		return true
+	}
+
+	client := docker.GetClient(DefaultVersion)
+	container := GetContainer(client, instance, false)
+	return isRunning(client, container)
+}
+
+func IsNoOp(data map[string]interface{}) bool {
+	b, ok := GetFieldsIfExist(data, "processData", "containerNoOpEvent")
+	if ok {
+		return InterfaceToBool(b)
+	}
+	return false
+}
+
+func GetContainer(client *client.Client, instance *model.Instance, byAgent bool) *types.Container {
+	if instance == nil {
+		return nil
+	}
+
+	// First look for UUID label directly
+	args := filters.NewArgs()
+	args.Add("label", fmt.Sprintf("%s=%s", UUIDLabel, instance.UUID))
+	options := types.ContainerListOptions{All: true, Filter: args}
+	labeledContainers, err := client.ContainerList(context.Background(), options)
+	if err == nil && len(labeledContainers) > 0 {
+		return &labeledContainers[0]
+	}
+
+	// Nest look by UUID using fallback method
+	options = types.ContainerListOptions{All: true}
+	containerList, err := client.ContainerList(context.Background(), options)
+	if err != nil {
+		return nil
+	}
+	container := findFirst(containerList, func(c *types.Container) bool {
+		if getUUID(c) == instance.UUID {
+			return true
+		}
+		return false
+	})
+
+	if container != nil {
+		return container
+	}
+	if externalID := instance.ExternalID; externalID != "" {
+		container = findFirst(containerList, func(c *types.Container) bool {
+			return idFilter(externalID, c)
+		})
+	}
+
+	if container != nil {
+		return container
+	}
+
+	if agentID := instance.AgentID; byAgent {
+		container = findFirst(containerList, func(c *types.Container) bool {
+			return agentIDFilter(strconv.Itoa(agentID), c)
+		})
+	}
+
+	return container
+
+}
+
+func isRunning(client *client.Client, container *types.Container) bool {
+	if container == nil {
+		return false
+	}
+	inspect, err := client.ContainerInspect(context.Background(), container.ID)
+	if err == nil {
+		return inspect.State.Running
+	}
+	return false
+}
+
+func getUUID(container *types.Container) string {
+	uuid, err := container.Labels[UUIDLabel]
```

Comment:
err is not the right name for this variable. `ok` is the conventional name and you can combine the get and if check like this:
```
if uuid, err := container.Labels[UUIDLabel]; ok {
  return uuid
}